batch_job/src/batch_job_mesos.c

```diff
@@ -0,0 +1,422 @@
+#include "batch_job.h"
+#include "batch_job_internal.h"
+#include "debug.h"
+#include "process.h"
+#include "macros.h"
+#include "stringtools.h"
+#include "path.h"
+#include "xxmalloc.h"
+#include "jx.h"
+#include "jx_print.h"
+#include "itable.h"
+#include "mesos_task.h"
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <errno.h>
+#include <signal.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <time.h>
+
+#define MAX_BUF_SIZE 4096
+#define FILE_TASK_INFO "mesos_task_info"
+#define FILE_TASK_STATE "mesos_task_state"
+#define MESOS_DONE_FILE "mesos_done"
+
+/*
+ * Principle of Operation:
+ * 
+ * Each time when makeflow submit a task, the information of the task is written into 
+ * mesos_task_info file. Meanwhile, the makeflow mesos scheduler keep polling 
+ * mesos_task_info, try to find if there is new task is ready to launch on mesos. 
+ * 
+ * After the task is complete on, the makeflow mesos scheduler writes the final state
+ * of the task to mesos_task_state. And the makeflow is keep monitoring this file and 
+ * collect all finished tasks.   
+ *
+ */
+
+static int counter = 0;
+static struct itable *finished_tasks = NULL;
+static int is_mesos_py_path_known = 0;
+static int is_mesos_master_known = 0;
+static int is_scheduler_running = 0;
+static const char *mesos_py_path = NULL;
+static const char *mesos_master = NULL;
+static pid_t mesos_scheduler_pid = 0;
+
+static void start_mesos_scheduler(struct batch_queue *q)
+{
+
+	pid_t mesos_pid;
+	mesos_pid = fork();			
+	mesos_scheduler_pid = mesos_pid;
+	char *mesos_cwd;
+	mesos_cwd = path_getcwd();
+
+	char exe_path[MAX_BUF_SIZE];
+
+	if(readlink("/proc/self/exe", exe_path, MAX_BUF_SIZE) == -1) {
+		debug(D_ERROR, "read \"proc/self/exe\" fail\n");
+		exit(EXIT_FAILURE);
+	}
+
+	char exe_dir_path[MAX_BUF_SIZE];
+	path_dirname(exe_path, exe_dir_path);
+
+    char *exe_py_path = string_format("%s/mf_mesos_scheduler", exe_dir_path);
+	char *envs[] = {"LD_PRELOAD=/afs/nd.edu/user37/ccl/software/external/gcc-4.9.3/amd64_linux26/lib64/libstdc++.so.6:/afs/nd.edu/user37/ccl/software/external/svn-1.9.4/amd64_linux26/lib/libsvn_delta-1.so", "CCTOOLS=/afs/crc.nd.edu/user/c/czheng2/cctools", NULL};
```

Comment:
Another place where we need some external configuration setting....

Or is this peculiar enough to ND that you could just set `LD_PRELOAD` before running your experiments?